src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -303,15 +315,29 @@ static Optional<float> GetVelocity(Unit* owner, Unit* target, G3D::Vector3 const
         }
 
         UnitMoveType moveType = Movement::SelectSpeedType(moveFlags);
-        speed = target->GetSpeed(moveType);
+        speed = std::max(target->GetSpeed(moveType), owner->GetSpeed(moveType));
+
         if (playerPet)
         {
-            float distance = owner->GetDistance2d(dest.x, dest.y) - target->GetObjectSize() - (*speed / 2.f);
+            float distance = owner->GetDistance2d(dest.x, dest.y) - (*speed / 2.f);
             if (distance > 0.f)
             {
                 float multiplier = 1.f + (distance / 10.f);
                 *speed *= multiplier;
             }
+            else
+            {
+                switch (moveType)
+                {
+                case MOVE_RUN_BACK:
+                case MOVE_SWIM_BACK:
+                case MOVE_FLIGHT_BACK:
+                    break;
+                default:
+                    *speed *= 0.9f;
+                    break;
+                }
+            }
```

Comment:
This will revert
https://github.com/azerothcore/azerothcore-wotlk/pull/16796

Reply:
Ah ye, my bad. I just copied the changes from my other branch, didnt think it would be outdated - it was a bit late 😛
Will fix this and double check that I didn't miss anything else 👌